api/src/users/controllers/users-controller.ts

```diff
@@ -49,24 +52,37 @@ class UsersController {
     }
 
     @ApiUnauthorizedResponse()
+    @ApiForbiddenResponse()
     @ApiOkResponse()
     @ApiNotFoundResponse()
     @UseGuards(JwtAuthGuard)
     @Get(":id")
-    async findOne(@Param("id") id: number) {
-        return this.userService.findOne(id);
+    async findOne(@Param("id") id: number, @UserReq("id") userId: number) {
+        return this.userService.findOneByIdAndUserJwtId(id, userId);
     }
 
     @ApiUnauthorizedResponse()
+    @ApiForbiddenResponse()
     @ApiOkResponse()
     @ApiNotFoundResponse()
     @UseGuards(JwtAuthGuard)
     @Delete(":id")
-    async delete(@Param("id") id: number) {
-        return this.userService.delete(id);
+    async delete(@Param("id") id: number, @UserReq("id") userId: number) {
+        const user = await this.userService.findOneByIdAndUserJwtId(id, userId);
+
+        const userAvatar = user.userAvatar;
```

Comment:
Why no 
```        if (user.userAvatar) {
            await this.userAvatarService.delete(
                user.userAvatar.id,
                user.userAvatar.sourcePath
            );
        }```
For me its 10x more readable.

Reply:
done